### PR TITLE
Ensure state is persisted

### PR DESCRIPTION
**Pull request by crosbymichael**
This also fixes a couple bugs with userns and adds some helpful features in nsinit.  Remove all the omitempty tags so that you can easily generate sample configs.

### CODE REVIEW

**Comment by LK4D4** [comment]
Hell, how I do that thing with labels :E

**Comment by mrunalp** [comment]
This branch is looking beautiful :)
Will review and try this out more tomorrow!

**Comment by dqminh** [comment]
> This branch is looking beautiful :)

+1. I love it.
Is there any ongoing process that attempts to integrate the api branch with docker's native driver ?

File: console.go [hunk 1]
```diff
@@ -2,7 +2,7 @@ package libcontainer
 
 import "io"
 
-// Console is a psuedo TTY.
+// Console represents a psuedo TTY.
```
Comment (thaJeztah) [inline console.go]: s/psuedo/pseudo/ :)

**Comment by vmarmol** [comment]
LGTM

File: nsinit/config.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/codegangsta/cli"
+	"github.com/docker/libcontainer/configs"
+)
+
+var createFlags = []cli.Flag{
+	cli.IntFlag{Name: "parent-death-signal", Usage: "set the signal that will be delivered to the process incase the parent dies"},
```
Comment (mrunalp) [inline nsinit/config.go]: s/incase/in case/

**Comment by crosbymichael** [comment]
Fixed the code review comments.

@dqminh yes, just try to get all the changes we want to make finalized and then i'll integrate it into docker.

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**